SignalServiceKit/src/Util/OWSBackgroundTask.m

```diff
@@ -76,36 +352,26 @@ - (void)dealloc
 
 - (void)startBackgroundTask
 {
-    // beginBackgroundTaskWithExpirationHandler must be called on the main thread.
     __weak typeof(self) weakSelf = self;
-    self.backgroundTaskId = [CurrentAppContext() beginBackgroundTaskWithExpirationHandler:^{
-        // Supposedly [UIApplication beginBackgroundTaskWithExpirationHandler]'s handler
-        // will always be called on the main thread, but in practice we've observed
-        // otherwise.  We use DispatchSyncMainThreadSafe() (note the sync) to ensure that
-        // this work is done on the main thread.
-        //
-        // See: https://developer.apple.com/documentation/uikit/uiapplication/1623031-beginbackgroundtaskwithexpiratio)
-        //
-        // Note the usage of OWSCAssert() to avoid capturing a reference to self.
-        OWSCAssert([NSThread isMainThread]);
-
-        DispatchSyncMainThreadSafe(^{
```

Comment:
We don't need to use `DispatchSyncMainThreadSafe()`.  See `[OWSBackgroundTaskManager 
 backgroundTaskExpired]`.